lib/THC/THCTensorIndex.cu

```diff
@@ -175,6 +175,26 @@ void THCudaTensor_indexFill(THCState *state, THCudaTensor *res_, int dim, THCuda
   THCudaCheck(cudaFree(stride_));
 }
 
+__global__ void THCudaTensor_kernel_indexSelect_contiguous(
```

Comment:
Could you add a comment explaining how you're mapping the work to the thread blocks? Will make it easier to maintain the code in future.